src/Type/Php/GetDefinedVarsFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use function array_filter;
+use function array_map;
+use function array_merge;
+use function array_values;
+use function count;
+use function range;
+
+// based on code by @ruudk
```

Comment:
Could you remove this
```suggestion
```

If you want to credit me, you can use GitHub's standard method for this, and append this to the commit body:

```
Co-Authored-By: Ruud Kamphuis <ruudk@users.noreply.github.com>
```

Reply:
ok